src/components/textbox/__next__/__internal__/input/input.style.ts

```diff
@@ -0,0 +1,123 @@
+import styled from "styled-components";
+import addFocusStyling from "../../../../../style/utils/add-focus-styling";
+
+interface InputContainerProps {
+  error?: boolean;
+  inputWidth?: number;
+  disabled?: boolean;
+  readOnly?: boolean;
+  size?: "small" | "medium" | "large";
+}
+
+const InputContainer = styled.div<InputContainerProps>`
+  display: flex;
+  ${({ inputWidth }) => inputWidth && `width: ${inputWidth}%;`}
+
+  min-height: ${({ size = "medium" }) => {
+    switch (size) {
+      case "small":
+        return "var(--global-size-s)";
+      case "large":
+        return "var(--global-size-l)";
+      default:
+        return "var(--global-size-m)";
+    }
+  }};
+
+  align-items: center;
+  align-self: stretch;
+  border-radius: var(--global-radius-action-m);
+
+  border: ${({ error, disabled, readOnly }) => {
+    if (disabled) {
+      return `var(--global-borderwidth-xs) solid var(--input-typical-border-disabled)`;
+    }
+    if (readOnly) {
+      return `var(--global-borderwidth-xs) solid var(--input-typical-border-read-only)`;
+    }
+    return error
+      ? `var(--global-borderwidth-s) solid var(--input-validation-border-error)`
+      : `var(--global-borderwidth-xs) solid var(--input-typical-border-default)`;
+  }};
+
+  background: ${({ disabled, readOnly }) => {
+    if (disabled) {
+      return "var(--input-typical-bg-disabled)";
+    }
+    if (readOnly) {
+      return "var(--input-typical-bg-read-only)";
+    }
+    return "var(--input-typical-bg-default)";
+  }};
+
+  &:focus-within {
+    ${addFocusStyling()}
+    z-index: 2;
+  }
+
+  [data-role="input-text-container"] {
+    display: flex;
+
+    padding: ${({ size = "medium" }) => {
+      switch (size) {
+        case "small":
+          return "0 var(--global-space-comp-s)";
```

Comment:
```suggestion
          return "var(--global-space-none) var(--global-space-comp-s)";
```

worth confirming with DS first though